File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        # Initialize the matrix as an empty list
        self.matrix = []

        # Split the input string by newline character to get rows
        for row in matrix_string.split('\n'):
            # Split each row by space and convert the elements to integers, then append to the matrix
            self.matrix.append([int(x) for x in row.split()])

    def row(self, index):
        # Return the specified row of the matrix (1-based index)
        return self.matrix[index - 1]

    def column(self, index):
        # Return the specified column of the matrix (1-based index)
        # Extract the elements at the specified column index from each row
        return [row[index - 1] for row in self.matrix]

File: javascript/matrix/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

// Define a class named 'Matrix'
export class Matrix {
  // Constructor for the Matrix class
  constructor(matrix) {
    // Initialize the 'matrix' property with the provided matrix data
    this.matrix = matrix;
  }

  // Define a 'rows' getter method to retrieve the matrix as rows
  get rows() {
    // Split the matrix string into an array of rows using newline character '\n',
    // then split each row into an array of numbers using space " " as a separator,
    // and map each value to a Number type to ensure they are numeric.
    return this.matrix.split('\n').map((row) => row.split(" ").map(Number));
  }

  // Define a 'columns' getter method to retrieve the matrix as columns
  get columns() {
    // Transpose the rows to columns by mapping each column index (i) to an array
    // that contains the values from that column across all rows.
    return this.rows[0].map((_, i) => this.rows.map((row) => row[i]));
  }
}
